nvflare/fuel/utils/fobs/fobs.py

```diff
@@ -143,6 +104,80 @@ def register(decomposer: Union[Decomposer, Type[Decomposer]]) -> None:
     _decomposers[name] = instance
 
 
+class Packer:
+    def __init__(self, manager: DatumManager):
+        self.manager = manager
+        self.enum_decomposer_name = _get_type_name(EnumTypeDecomposer)
+        self.data_decomposer_name = _get_type_name(DataClassDecomposer)
+
+    def pack(self, obj: Any) -> dict:
+
+        if type(obj) in MSGPACK_TYPES:
+            return obj
+
+        type_name = _get_type_name(obj.__class__)
+        if type_name not in _decomposers:
+            registered = False
+            if isinstance(obj, Enum):
+                if _enum_auto_registration:
+                    register_enum_types(type(obj))
+                    registered = True
+            else:
+                if callable(obj) or (not hasattr(obj, "__dict__")):
+                    raise TypeError(f"{type(obj)} can't be serialized by FOBS without a decomposer")
+                if _data_auto_registration:
+                    register_data_classes(type(obj))
+                    registered = True
+
+            if not registered:
+                return obj
+
+        decomposer = _decomposers[type_name]
+
+        decomposed = decomposer.decompose(obj, self.manager)
+        if self.manager:
+            decomposed = self.manager.externalize(decomposed)
+
+        return {FOBS_TYPE: type_name, FOBS_DATA: decomposed, FOBS_DECOMPOSER: _get_type_name(type(decomposer))}
+
+    def unpack(self, obj: Any) -> Any:
+
+        if type(obj) is not dict or FOBS_TYPE not in obj:
+            return obj
+
+        type_name = obj[FOBS_TYPE]
+        if type_name not in _decomposers:
+            registered = False
+            decomposer_name = obj.get(FOBS_DECOMPOSER)
+            cls = _load_class(type_name)
+            if not decomposer_name:
+                # Maintaining backward compatibility with auto enum registration
+                if _enum_auto_registration:
+                    if issubclass(cls, Enum):
+                        register_enum_types(cls)
+                        registered = True
```

Comment:
Do we need this part?
Users are writing their own decomposers, but we control both ends of pack <-> unpack right?
So this new Packer without this code, should already compatible with all existing decomposers?

Reply:
This is needed to unpack buffer generated by old version.